mbed/source/MbedBluetoothPlatform.cpp

```
@@ -0,0 +1,438 @@
+{
+    if (event->error) {
+        printError(event->error, "Error during the initialisation");
+        return;
+    }
+
+    print_mac_address();
+    if (_callback) {
+        call(_callback);
+    }
+}
+
+int MbedBluetoothPlatform::commonStartAdvertising()
+{
+    _is_scanner = false;
+    _is_connecting_or_syncing = false;
+
+    auto error = _ble.gap().startAdvertising(_adv_handle, _advertise_time);
+    if (error) {
+        printError(error, "Gap::startAdvertising() failed");
+    }
+
+    return error;
+}
+
+int MbedBluetoothPlatform::commonStartScan()
+{
+    _is_scanner = true;
+    _is_connecting_or_syncing = false;
+
+    ble::ScanParameters scan_params;
+    scan_params.setOwnAddressType(ble::own_address_type_t::RANDOM);
+
+    ble_error_t error = _ble.gap().setScanParameters(scan_params);
+    if (error) {
+        printError(error, "Gap::setScanParameters failed");
+        return error;
+    }
+
+    error = _ble.gap().startScan(_scan_time);
+    if (error) {
+        printError(error, "Gap::startScan failed");
+        return error;
+    }
+
+    auto eh = getEventHandler();
+    if (eh) {
+        eh->onScanStart(ScanStartEvent(_scan_time.valueInMs()));
+    }
+
+    return 0;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// BluetoothPlatform overrides
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+const char *MbedBluetoothPlatform::deviceName() const
+{
+    return "Power Consumption (mbed)";
+}
+
+void MbedBluetoothPlatform::init(const std::function<void()> &callback)
+{
+    _ble.gap().setEventHandler(this);
+    _callback = callback;
+    ble_error_t error = _ble.init(this, &MbedBluetoothPlatform::onInitComplete);
+    if (error) {
+        printError(error, "Error returned by BLE::init");
+        return;
+    }
+
+    _event_queue.dispatch_forever();
```

I'm not sure this should go here. Maybe you can return the error from the function and have a separate function for the dispatch.